src/ComposerIntegration/InstallAndBuildProcess.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Php\Pie\ComposerIntegration;
+
+use Composer\Package\CompletePackageInterface;
+use Composer\PartialComposer;
+use Php\Pie\Building\Build;
+use Php\Pie\DependencyResolver\Package;
+use Php\Pie\Downloading\DownloadedPackage;
+use Php\Pie\Installing\Install;
+
+use function sprintf;
+
+/** @internal This is not public API for PIE, so should not be depended upon unless you accept the risk of BC breaks */
+class InstallAndBuildProcess
+{
+    /** @psalm-suppress PossiblyUnusedMethod no direct reference; used in service locator */
+    public function __construct(
+        private readonly Build $pieBuild,
+        private readonly Install $pieInstall,
+        private readonly InstalledJsonMetadata $installedJsonMetadata,
+    ) {
+    }
+
+    public function __invoke(
+        PartialComposer $composer,
+        PieComposerRequest $composerRequest,
+        CompletePackageInterface $composerPackage,
+        string $installPath,
+    ): void {
+        $output = $composerRequest->pieOutput;
+
+        $downloadedPackage = DownloadedPackage::fromPackageAndExtractedPath(
+            Package::fromComposerCompletePackage($composerPackage),
```

Comment:
Note to self: check into missing ext name issue:

```
 Php\Pie\ExtensionName::normaliseFromString() at /pwd/src/ExtensionName.php:68
 Php\Pie\ExtensionName::determineFromComposerPackage() at /pwd/src/DependencyResolver/Package.php:64
 Php\Pie\DependencyResolver\Package::fromComposerCompletePackage() at /pwd/src/ComposerIntegration/InstallAndBuildProcess.php:36
 Php\Pie\ComposerIntegration\InstallAndBuildProcess->__invoke() at /pwd/src/ComposerIntegration/PiePackageInstaller.php:60
```

Reply:
Think the issue was out of date deps; https://github.com/composer/composer/commit/186d78cde397a3f4adbd600d61e5d4b85dbf7832 is needed for this branch to work; I do need to check if a tag has been released since :eyes:

Reply:
yes, we need min `^2.8.2` now

Reply:
Updated in b28d602